extensions/wikia/AdEngine/js/template/bfaaDesktop.js

```diff
@@ -72,24 +72,39 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 			recoveryTweaker.tweakSlot(params.slotName, iframe);
 		}
 
-		if (params.videoUrl && params.videoTriggerElement) {
-			slotTweaker.onReady(params.slotName, function() {
-				var divs = doc.querySelectorAll('#' + params.slotName + ' > div'),
-					imageContainer = divs[divs.length - 1],
-					video = uapVideoAd.init(doc.getElementById(params.slotName), imageContainer, params.videoUrl);
-
-				params.videoTriggerElement.addEventListener('click', function () {
-					uapVideoAd.playAndToggle(video, imageContainer);
-				});
+		if (videoEnabled(params)) {
+			videoAd.onLibraryReady(function () {
+				var playTrigger = videoAd.setupVideo(
+					adSlot.querySelector('div:last-of-type'),
+					document.body.clientWidth,
+					document.body.clientWidth / params.videoAspectRatio,
+					adSlot,
+					{
+						src: 'gpt',
+						slotName: params.slotName,
+						uap: params.uap
+					}
+				);
+
+				params.videoTriggerElement.addEventListener('click', playTrigger);
 			});
 		}
 	}
 
+	function videoEnabled(params) {
+		return params.videoTriggerElement && params.videoAspectRatio;
+	}
+
 	function show(params) {
+		adSlot = doc.getElementById(params.slotName);
 		nav = doc.getElementById('globalNavigation');
 		page = doc.getElementsByClassName('WikiaSiteWrapper')[0];
 		wrapper = doc.getElementById('WikiaTopAds');
 
+		if (videoEnabled(params)) {
+			videoAd.init();
```

Comment:
Does it affect BFAA load time/UX? Can we move it to https://github.com/Wikia/app/pull/11744/files#diff-390793e4ffe62f6a1599cf70d636f0bdR75 ?